api/src/main/java/org/apache/iceberg/types/IndexByName.java

```diff
@@ -128,9 +157,15 @@ private void addField(String name, int fieldId) {
     }
 
     Integer existingFieldId = nameToId.put(fullName, fieldId);
-    if (existingFieldId != null && !"element".equals(name) && !"value".equals(name)) {
-      throw new ValidationException(
-          "Invalid schema: multiple fields for name %s: %s and %s", fullName, existingFieldId, fieldId);
+    ValidationException.check(existingFieldId == null,
+        "Invalid schema: multiple fields for name %s: %s and %s", fullName, existingFieldId, fieldId);
+
+    // if the short name is not
```

Comment:
unfinished comment?

Reply:
If the short name is not empty and we haven't already added an identical short name?